metagraph/src/kmer/kmer_extractor.cpp

```diff
@@ -399,6 +399,44 @@ void KmerExtractorBOSS::sequence_to_kmers(std::string_view,
                                           Vector<Kmer256>*,
                                           bool);
 
+template <typename KMER>
+KMER KmerExtractorBOSS::sequence_to_kmer(std::string_view sequence,
+                                         bool canonical_mode) {
+    // encode sequence
+    std::vector<TAlphabet> seq(sequence.size(), alphabet.size());
+
+    std::transform(sequence.begin(), sequence.end(), seq.begin(), [](char c) {
+        return c != alphabet[0] ? encode(c) : 0;
+    });
+
+    if (std::find(seq.begin(), seq.end(), alphabet.size()) != seq.end())
+        throw std::runtime_error("Invalid characters encoded");
```

Comment:
I don't get it. It will never throw, you map '$' to 0 and then check for `alphabet.size()`

Reply:
oops, I forgot to remove this check...

Reply:
If the input sequence has non-ASCII characters, this can still throw. This should never happen in my usage of this function, but I kept the throw here just in case.